### PR TITLE
Yuchen/set tmp,two nodes

### PR DESCRIPTION
**Pull request by YuchenMu**
"set_tmp" new API, new tests and new documentation done.

### CODE REVIEW

**Comment by szaldivar** [comment]
I will wait for the two nodes PR to be merged so I do not repeat myself here

**Comment by YuchenMu** [comment]
> I will wait for the two nodes PR to be merged so I do not repeat myself here

I will update two nodes after the meeting

**Comment by szaldivar** [comment]
you might want to merge the changes you did in your other branch to fix the merge conflicts here

**Comment by YuchenMu** [comment]
Yes, that’s what I’m doing now. Thank you!

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:12]: maybe we need descriptions for these properties?

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:77]: don't set the default value in the parameters list, and detect if they are undefined

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:80]: same as previous here

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:122]: Just as a heads up, you can use this syntax to deconstruct values from the units_converter:
```javascript
({tmp: set_tmp} = units_converter({tmp: set_tmp}, "SI"))
```
that will do the same as you do inside this if but you don't have to change it. Just wanted to mentioned it

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:126]: are you sure the `check_standard_compliance` and `check_standard_compliance_array` check the same limits?

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:258]: this breaks if the arrays are not of length 1

Comment (szaldivar) [reply src/models/set_tmp.js:258]: that value doesn't do anything in `two_nodes` can you check the implementation of two_nodes pls

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:287]: omit all this and keep just keep the for loop

**Review by szaldivar — CHANGES_REQUESTED** [review]

File: src/models/set_tmp.js
Comment (YuchenMu) [reply src/models/set_tmp.js:144]: In the Python version, I can only tell it is checking "ashrae". I did not see other examples in the documentation of Python version

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:84]: assign this value depending on the unit system so you avoid doing it below

Comment (szaldivar) [reply src/models/set_tmp.js:84]:
```javascript
body_surface_area = units === "SI" ? 1.8258 : 19.65;
```

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:100]: remove this and do it in the undefined checks above

File: src/models/set_tmp.js
Comment (szaldivar) [reply src/models/set_tmp.js:103]: remove this and do the same as my suggestion for `body_surface_area`

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:158]: these are arrays now, you need to index into them, do `isNaN(tdb_valid[0])`

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:262]: same comment as the other function

File: src/models/two_nodes.js
Comment (szaldivar) [inline src/models/two_nodes.js:145]: keep the default values in the parameters here.
Maybe you did not understand why in the other function it was better to leave them to undefined
In the other function you needed two different default values depending in the unit system, so it is better to leave it to undefined and then assign them based in the unit system like
```javascript
some_var = units === "SI" ? si_default_value : ip_default_value
```
In here there is no point in doing it this way, it is way better to set in it in the parameters

**Review by szaldivar — CHANGES_REQUESTED** [review]

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:233]: move to parameters

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:237]: move to parameters

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:248]: you need this to be an array

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:249]: needs to be an array right?

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:241]: what if units are `"IP"` ?

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:245]: what if units are `"IP"` ?

**Review by szaldivar — CHANGES_REQUESTED** [review]

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:42]:
```suggestion
 * @param {number} [body_surface_area] Body surface area, default value 1.8258 [m2] in [ft2] if units = ‘IP’
```

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:43]:
```suggestion
 * @param {number} [p_atm] Atmospheric pressure, default value 101325 [Pa] in [atm] if units = ‘IP’
```

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:122]:
```suggestion
    } = check_standard_compliance_array("ASHRAE", {
```

File: src/models/set_tmp.js
Comment (szaldivar) [inline src/models/set_tmp.js:269]:
```suggestion
    } = check_standard_compliance_array("ASHRAE", {
```

**Review by szaldivar — APPROVED** [review]

**Pull request merged**
